packages/caver-account/src/accountKey/accountKeyWeightedMultiSig.js

```diff
@@ -0,0 +1,117 @@
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const RLP = require('eth-lib/lib/rlp')
+const Bytes = require('eth-lib/lib/bytes')
+const WeightedPublicKey = require('./weightedPublicKey')
+const utils = require('../../../caver-utils')
+const { ACCOUNT_KEY_TAG } = require('./accountKeyHelper')
+
+/**
+ * Representing an AccountKeyWeightedMultiSig.
+ * @class
+ */
+class AccountKeyWeightedMultiSig {
+    /**
+     * Decodes an RLP-encoded AccountKeyWeightedMultiSig string.
+     * @param {string} rlpEncodedKey - An RLP-encoded AccountKeyWeightedMultiSig string.
+     * @return {AccountKeyWeightedMultiSig}
+     */
+    static decode(rlpEncodedKey) {
+        rlpEncodedKey = utils.addHexPrefix(rlpEncodedKey)
+        if (!rlpEncodedKey.startsWith(ACCOUNT_KEY_TAG.ACCOUNT_KEY_WEIGHTED_MULTISIG_TAG))
+            throw new Error(`Cannot decode to AccountKeyWeightedMultiSig: ${rlpEncodedKey}`)
+
+        const [threshold, multiSigkeys] = RLP.decode(`0x${rlpEncodedKey.slice(ACCOUNT_KEY_TAG.ACCOUNT_KEY_WEIGHTED_MULTISIG_TAG.length)}`)
+        const weightedPublicKeys = multiSigkeys.map(wieghtedPublicKey => {
+            return new WeightedPublicKey(wieghtedPublicKey[0], wieghtedPublicKey[1])
+        })
+        return new AccountKeyWeightedMultiSig(threshold, weightedPublicKeys)
+    }
+
+    /**
+     * Creates an instance of AccountKeyWeighedMultiSig.
+     * @param {Array.<string>} publicKeyArray - An array of public key strings.
+     * @param {object} options - An options which defines threshold and weight.
+     * @return {AccountKeyWeightedMultiSig}
+     */
+    static fromPublicKeysAndOptions(publicKeyArray, options) {
+        const weightedPublicKeys = []
+        for (let i = 0; i < publicKeyArray.length; i++) {
+            const weightedPublicKey = new WeightedPublicKey(options.weight[i], publicKeyArray[i])
+            weightedPublicKeys.push(weightedPublicKey)
+        }
+        return new AccountKeyWeightedMultiSig(options.threshold, weightedPublicKeys)
+    }
+
+    /**
+     * Create an instance of AccountKeyWeightedMultiSig.
+     * @param {number} threshold - The threshold of accountKey.
+     * @param {Array.<WeightedPublicKey>} weightedPublicKeys - An array of instances of WeightedPublicKeys
+     */
+    constructor(threshold, weightedPublicKeys) {
+        this._threshold = utils.hexToNumber(threshold)
+        this._weightedPublicKeys = weightedPublicKeys || []
```

Comment:
Don't we need to check the type of `weightedPublicKeys`?

Reply:
I added check logic in here.